leakcanary-android-core/src/main/java/leakcanary/internal/activity/db/LeaksDbHelper.kt

```diff
@@ -54,11 +54,12 @@ internal class LeaksDbHelper(context: Context) : SQLiteOpenHelper(
               val analysis = pair.second as HeapAnalysisSuccess
 
               val unreachableObjects = try {
-                analysis.unreachableObjects
+                // Compiler doesn't know it but runtime can have null.
+                analysis.unreachableObjects.orEmpty()
               } catch (ignored: NullPointerException) {
                 // This currently doesn't trigger but the Kotlin compiler might change one day.
                 emptyList()
-              } ?: emptyList() // Compiler doesn't know it but runtime can have null.
+              }
```

Comment:
Is there a deeper reason or solution for this NPE ?

Reply:
I think this had to do with java deserialization and the fact that it can hydrate objects with null fields even though they're not kotlin nullable, and there won't be any null checks until you assign / pass the returned value.